src/commands/commerce/extension/register.ts

```diff
@@ -0,0 +1,140 @@
+                this.flags['apex-classname'],
+                this.org.getUsername()
+            ),
+        };
+    }
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public registerApex(
+        storeRegisteredName: string,
+        storeEPN: string,
+        storeApexClass: string,
+        storeUserName: string
+    ): string {
+        // get apexClass id from table
+        const apexClassId = this.getApexClass(storeApexClass, storeUserName);
+
+        // checks if user passed valid EPN
+        const epnVal = forceDataSoql(
+            `SELECT Value FROM PicklistValueInfo WHERE Value='${storeEPN}' AND EntityParticle.DurableId = 'RegisteredExternalService.ExtensionPointName' LIMIT 1`,
+            storeUserName
+        );
+        if (epnVal.result.totalSize === 0) {
+            throw new SfdxError(msgs.getMessage('extension.register.errEPN'));
+        }
+
+        // Inserts 5 fields into registeredExternalService table
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const results = forceDataRecordCreate(
+            'RegisteredExternalService',
+            `DeveloperName=${storeRegisteredName} MasterLabel=${storeRegisteredName} ExtensionPointName=${storeEPN} ExternalServiceProviderId=${apexClassId} ExternalServiceProviderType='Extension'`,
+            storeUserName
+        );
+        // Validates/checks for already existing unique name?
+        if (results instanceof SfdxError && results.message.indexOf('DUPLICATE_VALUE') > 0) {
+            throw new SfdxError(msgs.getMessage('extension.register.nameAlreadyExists', [storeRegisteredName]));
+        }
+        // JSON response of inserted record
+        return this.getInsertedRecord(storeRegisteredName, storeApexClass, storeUserName);
+    }
+
+    // Queries for apexclassId from existing apexClass table
+    private getApexClass(storeApexClass: string, storeUserName: string): string {
+        const QUERY_GET_APEX_CLASS = `SELECT Id FROM ApexClass WHERE Name='${storeApexClass}' LIMIT 1`;
+        let apexClassId: string;
+        try {
+            apexClassId = forceDataSoql(QUERY_GET_APEX_CLASS, storeUserName).result.records[0].Id;
+        } catch (e) {
+            throw new SfdxError(msgs.getMessage('extension.register.errApexClass', [storeApexClass]));
+        }
+        return apexClassId;
+    }
+    // returns entire record from RegisteredExternalService in JSON format
+    private getInsertedRecord(storeRegisteredName: string, storeApexClass: string, storeUserName: string): string {
+        const RegisteredTable = forceDataSoql(
+            `SELECT Id,ConfigUrl,DeveloperName,DocumentationUrl,ExtensionPointName,ExternalServiceProviderId,ExternalServiceProviderType,Language,MasterLabel,NamespacePrefix from RegisteredExternalService WHERE DeveloperName='${storeRegisteredName}'`,
+            storeUserName
+        );
+
+        for (const element of RegisteredTable.result.records) {
```

Comment:
I don't think we should add anything that is not part of the query output. It is not a standard approach and is not easy to maintain.
I would suggest getting rid of `ApexClassName`